mitodl/course-search-utils
# Include zero count active facets in facet options

### What are the relevant tickets?
For https://github.com/mitodl/mit-open/pull/675

### Description (What does it do?)
- In https://github.com/mitodl/mit-open/pull/675, I added code in MIT Open to append zero-count active facets to the facet options. This PR moves that to course-search-utils.
- Rewrites the facet display tests using testing library instead of Enzyme

### Screenshots (if appropriate):

Note the zero-count active checkbox

![Screenshot 2024-03-27 at 11 08 28 AM](https://github.com/mitodl/course-search-utils/assets/9010790/c5002041-e726-4f1b-820e-b01338a46612)

### How can this be tested?

Using https://github.com/mitodl/mit-open/pull/675, activate a search filter then EITHER:
- add a text query that makes the filter have zero results
- OR switch to one of the resource tabs with zero results
The active filter should still show in the facet options list.
